src/main/java/com/nnaltakyan/core/auth/domain/verification/events/SendVerificationEmailEvent.java

```diff
@@ -0,0 +1,15 @@
+package com.nnaltakyan.core.auth.domain.verification.events;
+
+import com.nnaltakyan.core.auth.domain.auth.service.AuthenticationService;
+import lombok.Getter;
+import org.springframework.context.ApplicationEvent;
+
+@Getter
+public class SendVerificationEmailEvent extends ApplicationEvent {
+
+    private final Long userId;
+    public SendVerificationEmailEvent(AuthenticationService source, Long userId) {
```

Comment:
I don't think that there is a compulsion to use AuthenticationService

Reply:
@nareknaltakyan1 we need to define the event source which is EventPublisher class in our case. I made according changes. Please take a look.